### PR TITLE
Memory Pair Game 

### PR DESCRIPTION
**Pull request by YuliiaHordiichuk**
### Memory Pair Game 

[Demo](https://yuliiahordiichuk.github.io/memory-game/)
[Code base](https://github.com/YuliiaHordiichuk/memory-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/YuliiaHordiichuk/memory-game/index.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <title>Memory Pair Game</title>
+</head>
+
+<body>
+    <ul class='cards'>
```
Comment (AMashoshyna) [inline submissions/YuliiaHordiichuk/memory-game/index.html:12]: Mixing single quotes and double quotes without reason is better to be avoided. Usually we use double quotes in html and single quotes in js

File: submissions/YuliiaHordiichuk/memory-game/index.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+class Game {
```
Comment (AMashoshyna) [inline submissions/YuliiaHordiichuk/memory-game/index.js:1]: Why did you decide to use a class here? As opposed to, for example, regular function

File: submissions/YuliiaHordiichuk/memory-game/index.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    constructor() {
+        this.cardsContainer = document.querySelector('.cards');
```
Comment (AMashoshyna) [inline submissions/YuliiaHordiichuk/memory-game/index.js:3]: Let's check what is really needed as a class member and what can be saved as a constant outside of class

File: submissions/YuliiaHordiichuk/memory-game/index.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            <li class='card'>
+                <button class='card__btn' data-id='${card}'>
+                    <img class='card__front' src='assets/cards-front.jpg' alt='card-front'>
+                    <img class='card__back' src='assets/${card}.jpg' alt='card-back'>
+                </button>
+            </li>`
+        }).join(''); 
+    }; 
+
+    addClickHendler () {
```
Comment (AMashoshyna) [inline submissions/YuliiaHordiichuk/memory-game/index.js:37]: looks like a typo: hEndler => hAndler

File: submissions/YuliiaHordiichuk/memory-game/index.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        this.cardsContainer.addEventListener('click', ({ target: button }) => {
+            const buttonId = button.dataset.id;
+
+            if(buttonId) {
+                button.classList.add('active');
```
Comment (AMashoshyna) [inline submissions/YuliiaHordiichuk/memory-game/index.js:42]: Again mix of single quotes and double quotes. No matter what you choose, just be consistent.

File: submissions/YuliiaHordiichuk/memory-game/index.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                if (this.openCards.length === this.oneMatchCards) {
+                    this.checkCards(); 
+                }; 
+            }  
+        });   
+    }
+
+    checkCards () {
+        if (this.openCards[0].dataset.id === this.openCards[1].dataset.id) {
+            const cardsToHide = [].concat(this.openCards);
```
Comment (AMashoshyna) [inline submissions/YuliiaHordiichuk/memory-game/index.js:58]: I'm assuming that you want to make copy of an array here. If so, you can use[ spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_array_literals)

File: submissions/YuliiaHordiichuk/memory-game/index.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+    mixCards () {
+        return this.cardsBack.concat(this.cardsBack).sort(() =>  0.5 - Math.random());  
+    }
+
+    createCards () {
+        return this.mixCards().map(card => {
+            return `
+            <li class='card'>
+                <button class='card__btn' data-id='${card}'>
```
Comment (AMashoshyna) [inline submissions/YuliiaHordiichuk/memory-game/index.js:29]: Why do you need a button here?

**Review by AMashoshyna — COMMENTED** [review]
Good job! Please take a look at my comments

File: submissions/YuliiaHordiichuk/memory-game/index.js (hunk 7 above)
Comment (YuliiaHordiichuk) [reply submissions/YuliiaHordiichuk/memory-game/index.js:29]: I use the button in this case to provide keyboard accessibility

File: submissions/YuliiaHordiichuk/memory-game/index.js (hunk 2 above)
Comment (YuliiaHordiichuk) [reply submissions/YuliiaHordiichuk/memory-game/index.js:1]: I've got some knowledge about classes and wanted to practice it. Should I change it to a regular function?

Comment (AMashoshyna) [reply submissions/YuliiaHordiichuk/memory-game/index.js:8]: No, it's ok, just wanted to know the reasoning

**Comment by AMashoshyna** [comment]
Well done!

**Pull request merged**